### PR TITLE
Enable mavproxy to watch multiple messages

### PR DESCRIPTION
**Pull request by patrickelectric**
Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

**Comment by peterbarker** [comment]
Sorry, I haven't tried to run this.  But where are the changes which makes this a dict?

How do you "unwatch" something?  That's currently done with "watch" bare.

**Comment by patrickelectric** [comment]
Is not a dict, is a list with the messages names.
To 'unwatch', just uses watch as a normal case.

**Comment by OXINARF** [comment]
We should probably get out of the loop once the message is printed right? We don't need to print it multiple times.

**Comment by patrickelectric** [comment]
> We should probably get out of the loop once the message is printed right? We don't need to print it multiple times.

that's what `status` command do.

**Comment by OXINARF** [comment]
@patrickelectric What I meant is that it's looping through the entire watch list, even after it has been printed. You can have two watch types that will match the same message but you don't want to print it twice, right?

**Comment by patrickelectric** [comment]
@OXINARF check now, now it'll break if the message is find.

**Comment by patrickelectric** [comment]
Any update ?

**Comment by patrickelectric** [comment]
72 days and counting..

File: MAVProxy/mavproxy.py [hunk 1]
```diff
@@ -561,11 +561,12 @@ def process_mavlink(slave):
         return
     if msgs is None:
         return
-    if mpstate.settings.mavfwd and not mpstate.status.setup_mode:
-        for m in msgs:
-            if mpstate.status.watch is not None:
-                if fnmatch.fnmatch(m.get_type().upper(), mpstate.status.watch.upper()):
+    if mpstate.settings.mavfwd and not mpstate.status.setup_mode and mpstate.status.watch:
+        for msg_type in mpstate.status.watch:
+            for m in msgs:
```
Comment (OXINARF) [inline MAVProxy/mavproxy.py]: I think this still isn't OK. The for loops should be switched. What we want is for all the messages we have received (`msgs`), check if it matches at most one watch regex that has been set - what it does now is for every regex that has been set check if there is at most one received message that matches.

Comment (patrickelectric) [reply MAVProxy/mavproxy.py]: This way will avoid unnecessary loop interactions.
I did a python script if you want to try :)

**Comment by patrickelectric** [comment]
@OXINARF any update ?

**Pull request merged**